R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+#' Internal function to extract summary output from survival models.
+#'
+#' @param model survival object with model
+#' @param start_cohort start date of the study
+#' @param end_cohort end date of the study
+#' @return Data frame with survival data
+#' @export
+extract_surv_model <- function(model, start_cohort, end_cohort) {
+  days <- end_cohort - start_cohort
+  tte <- seq(0, as.numeric(days) - 1, by = 1)
+  res <- summary(model, times = tte, scale = 1)
+  cols <- lapply(c(2:6, 8:16), function(x) res[x])
+  tbl <- do.call(data.frame, cols)
+  tbl$date <- tbl$time + start_cohort
+  return(tbl)
+}
```

Comment:
Yes, I will make it internal. 

Apparently, there's no standard method to extract data from survival objects.  It can be made in different ways and using other libraries. Anyway, I will use the names of the columns.

Reply:
Resolved in https://github.com/epiverse-trace/vaccineff/pull/31/commits/e2d519efd7175e141e4a64cb188f001f64ef9650